Metalama.Framework.Engine/Pipeline/SourceTransformer.cs

```diff
@@ -54,9 +54,9 @@ public sealed class SourceTransformer : ISourceTransformerWithServices
                 DotNetSdkDirectory = dotNetSdkDirectory
             };
 
-            BackstageServiceFactoryInitializer.Initialize( backstageOptions );
-
-            var serviceProvider = BackstageServiceFactory.ServiceProvider;
+            // We don't use BackstageServiceFactory.ServiceProvider here, because it's lifetime goes over the lifetime of a source transformer,
+            // and we manage disposal of services at the end of the source transformer's lifetime.
```

Comment:
Would it make sense to have a way of disposing the whole service provider, so that even services that are not directly used here can be disposed?

Reply:
I don't think it's worth it, as most of the services don't need it.